File: src/main/java/com/jakeesveld/zoos/model/Animal.java
package com.jakeesveld.zoos.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "animal")
public class Animal {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long animalid;

    private String animaltype;

    @ManyToMany(mappedBy = "animalList")
    private List<Zoo> zoos = new ArrayList<>();

    public Animal() {
    }

    public Animal(String animaltype) {
        this.animaltype = animaltype;
    }

    public Long getAnimalid() {
        return animalid;
    }

    public void setAnimalid(Long animalid) {
        this.animalid = animalid;
    }

    public String getAnimaltype() {
        return animaltype;
    }

    public void setAnimaltype(String animaltype) {
        this.animaltype = animaltype;
    }

    public List<Zoo> getZoos() {
        return zoos;
    }

    public void setZoos(List<Zoo> zoos) {
        this.zoos = zoos;
    }
}

File: src/main/java/com/jakeesveld/zoos/model/Animal.kt
package com.jakeesveld.zoos.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

import javax.persistence.*
import java.util.ArrayList

@Entity
@Table(name = "animal")
class Animal {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var animalid: Long? = null

    var animaltype: String? = null

    @ManyToMany(mappedBy = "animalList")
    @JsonIgnoreProperties("animalList")
    var zoos: MutableList<Zoo> = mutableListOf()

    constructor() {}

    constructor(animaltype: String) {
        this.animaltype = animaltype
    }
}


